src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -1562,146 +1544,6 @@ private BoundExpression GetDefaultParameterSpecialNoConversion(SyntaxNode syntax
                 argsToParamsBuilder.AddRange(argsToParamsOpt);
             }
 
-            BoundExpression? collection = null;
-
-            if (expanded)
-            {
-                Debug.Assert(paramsArgsBuilder is not null);
-                ImmutableArray<BoundExpression> collectionArgs = paramsArgsBuilder.ToImmutableAndFree();
-                int collectionArgsLength = collectionArgs.Length;
-
-                TypeSymbol collectionType = parameters[paramsIndex].Type;
```

Comment:
>TypeSymbol collectionType = parameters[paramsIndex].Type; [](http://example.com/codeflow?start=16&length=57)

Code starting from here through ```Debug.Assert(collection.IsParamsArrayOrCollection);``` is extracted to ```CreateParamsCollection``` helper. No chnages to the code other than making  ```parameters[paramsIndex]``` a parameter named  "paramsParameter".